### ISSUE TITLE
SC2086 find with multiple folders should not be raised

### ISSUE THREAD

**Issue by PaulaAlessio** [koalaman/shellcheck#1962]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): 0.3.8 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
myfolders="A B"
find ${myfolders[@]} type -f 

```

#### Here's what shellcheck currently says:
Double quote to prevent globbing and word splitting. [SC2086]

#### Here's what I wanted or expected to see:
Nothing

The problem is that if we double quote the variable myfolders, then find does not interpret the folders as multiple folders and says: 
find: ‘A B’: No such file or directory

**Comment by furkanpham** [comment]
`"${myfolders[@]}"` expands to a list of words, with each element in the myfolders array as one word. The reason why it does not interpret "A B" as two seperate folders is because `myfolders` isn't an array but a string, so there's nothing to expand since only one element exists. Creating an array using `=()` syntax will work as intended.
```bash
#!/bin/bash
myfolders=("A" "B" "C D")
find "${myfolders[@]}" -type f
```
This also makes it possible to include directories with whitespace (like `C D`) without relying on word splitting.

**Comment by PaulaAlessio** [comment]
Understood, 

many thanks!
